examples/cookbook/architecture/command/lib/command.dart

```diff
@@ -0,0 +1,104 @@
+// Copyright 2024 The Flutter team. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:flutter/foundation.dart';
+
+import 'result.dart';
+
+/// Defines a command action that returns a [Result] of type [T].
+/// Used by [Command0] for actions without arguments.
+typedef CommandAction0<T> = Future<Result<T>> Function();
+
+/// Defines a command action that returns a [Result] of type [T].
+/// Takes an argument of type [A].
+/// Used by [Command1] for actions with one argument.
+typedef CommandAction1<T, A> = Future<Result<T>> Function(A);
+
+/// Facilitates interaction with a view model.
+///
+/// Encapsulates an action,
+/// exposes its running and error states,
+/// and ensures that it can't be launched again until it finishes.
+///
+/// Use [Command0] for actions without arguments.
+/// Use [Command1] for actions with one argument.
+///
+/// Actions must return a [Result] of type [T].
+///
+/// Consume the action result by listening to changes,
+/// then call to [clearResult] when the state is consumed.
+abstract class Command<T> extends ChangeNotifier {
+  bool _running = false;
+
+  /// Whether the action is running.
+  bool get running => _running;
+
+  Result<T>? _result;
+
+  /// Whether the action completed with an error.
+  bool get error => _result is Error;
+
+  /// Whether the action completed successfully.
+  bool get completed => _result is Ok;
+
+  /// The result of the most recent action.
+  ///
+  /// Returns `null` if the action is running or completed with an error.
+  Result? get result => _result;
```

Comment:
Should this be `Result<T>`?

```suggestion
  Result? get result => _result;
```

Reply:
Yes, good catch!